apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/grpc/StargateBridgeClientImpl.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.stargate.sgv2.api.common.grpc;
+
+import io.smallrye.mutiny.Uni;
+import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.sgv2.api.common.StargateRequestInfo;
+import io.stargate.sgv2.api.common.schema.SchemaManager;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.apache.commons.codec.binary.Hex;
+
+@ApplicationScoped
+public class StargateBridgeClientImpl implements StargateBridgeClient {
+
+  private static final Function<String, Uni<? extends Schema.CqlKeyspaceDescribe>>
+      MISSING_KEYSPACE = ks -> Uni.createFrom().failure(new UnauthorizedKeyspaceException(ks));
+
+  @Inject SchemaManager schemaManager;
+
+  @Inject StargateRequestInfo requestInfo;
+
+  @Override
+  public CompletionStage<QueryOuterClass.Response> executeQueryAsync(QueryOuterClass.Query query) {
+    return requestInfo.getStargateBridge().executeQuery(query).subscribeAsCompletionStage();
+  }
+
+  @Override
+  public CompletionStage<QueryOuterClass.Response> executeQueryAsync(
+      String keyspaceName,
+      String tableName,
+      Function<Optional<Schema.CqlTable>, QueryOuterClass.Query> queryProducer) {
+
+    // TODO implement optimistic queries (probably requires changes directly in SchemaManager)
+    return getTableAsync(keyspaceName, tableName, true)
+        .thenCompose(table -> executeQueryAsync(queryProducer.apply(table)));
```

Comment:
I'm not sure how to implement that. It requires access to Caffeine's `getIfPresent()`, but it doesn't look like quarkus-cache exposes anything like it.

Reply:
you can use `keyspaceCache.as(CaffeineCache.class)`, that gives you the access to all keys, so you continue from there.. 

Yes, quite inconvenient, let's open a Quarkus ticket, maybe even contribute this quickly..

Reply:
I've created #1977 as a follow-up, so that it doesn't block this PR.